worker-workflow/src/main/resources/workflow-control.js

```diff
@@ -193,6 +193,47 @@ function applyActionDetails(document, actionDetails, terminateOnFailure) {
     }
 }
 
+function onAfterProcessDocument(e){
+    haveFailuresChanged(e.document);
+}
+
+function haveFailuresChanged(document) {
+    if (document.getFailures().isChanged()) {
+
+        var listOfFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfFailures.add(failure);
+        });
+
+        document.getFailures().reset();
+
+        var listOfOriginalFailures = new java.util.ArrayList();
+        document.getFailures().stream().forEach(function (failure) {
+            listOfOriginalFailures.add(failure);
+        });
+
+        var newListOfFailures = new java.util.ArrayList();
+        for each (var f in listOfFailures) {
+            if (!listOfOriginalFailures.contains(f)) {
```

Comment:
I'm surprised just doing the comparison in as simple a way that as that works.  The `equals()` method hasn't been implemented in [FailureImpl.java](https://github.com/CAFDataProcessing/worker-document/blob/v4.1.0/worker-document/src/main/java/com/hpe/caf/worker/document/impl/FailureImpl.java) so I would have thought that was just comparing object identities, which I think will be different so I'm not sure how this could be working.